docs/internal.md

```diff
@@ -0,0 +1,71 @@
+# Internal of Azure SignalR Service
+
+- [Server Connections](#server-connections)
+- [Client Connections](#client-connections)
+- [Transport Data between Client and Server](#transport)
+
+Azure SignalR Service is built on top of ASP.NET Core SignalR framework.
+You can start from a local ASP.NET Core SignalR application, then seamlessly move to Azure SignalR Service with a few lines of code change.
+
+The following diagram shows the typical architecture when you use Azure SignalR Service with your own application server.
+Behavior of clients and servers will be different from self-hosting ASP.NET Core SignalR.
+More details are explained in below sections.
+
+![Architecture](./images/arch.png)
+
+## Server Connections
+
+When self-hosting ASP.NET Core SignalR, application server will listen for client connections and serve them directly.
+
+With SignalR Service, application server will not accept SignalR client connections anymore.
+A `negotiate` endpoint will be exposed by Azure SignalR Service SDK for each hub.
+It will respond to clients' negotiate requests and redirect them to SignalR Service.
+Eventually, all clients will be connected to SignalR Service.
+Read more about it in [Client Connections](#client-connections) section.
+
+Right after application server is started, a few WebSocket connections are initiated to SignalR Service.
+By default the connection count is 5 and it is configurable.

```

Comment:
After application server is started, the Azure SignalR SDK will open 5 WebSocket connections to the SignalR Service. The number of used connections is configurable.